src/util/mod.rs

```diff
@@ -180,7 +180,10 @@ pub fn escape(data: &[u8]) -> String {
             b'\t' => escaped.extend_from_slice(br"\t"),
             b'"' => escaped.extend_from_slice(b"\\\""),
             b'\\' => escaped.extend_from_slice(br"\\"),
-            b'\x20'...b'\x7e' => escaped.push(c),
+            b'\x20' |
+            b'\x21' |
```

Comment:
I think the correctness should be guaranteed by the tests, rather than keeping matching arm non-overlapping, which is not very readable or efficient as we can see.